app/data/views.py

```diff
@@ -229,3 +235,48 @@ def get_serializer_class(self):
 
 class DriverBoundaryViewSet(BoundaryViewSet):
     permission_classes = (IsAdminOrReadOnly,)
+
+
+class RecordCsvExportViewSet(viewsets.ViewSet):
+    """A view for interacting with CSV export jobs
+
+    Since these jobs are not model-backed, we won't use any of the standard DRF mixins
+    """
+    permissions_classes = (IsAdminOrReadOnly,)
+
+    def retrieve(self, request, pk=None):
+        """Return the status of the celery task with query_params['taskid']"""
+        # N.B. Celery will never return an error if a task_id doesn't correspond to a
+        # real task; it will simply return a task with a status of 'PENDING' that will never
+        # complete.
+        job_result = export_csv.AsyncResult(pk)
+        if job_result.state in states.READY_STATES:
+            if job_result.state in states.EXCEPTION_STATES:
+                e = job_result.get(propagate=False)
+                return Response({'status': job_result.state, 'error': str(e)})
+            # Set up the URL to proxy to the celery worker
+            # TODO: This won't work with multiple celery workers
+            # TODO: We should add a cleanup task to prevent result files from accumulating
+            # on the celery worker.
+            uri = '{scheme}://{host}{prefix}{file}'.format(scheme=request.scheme,
+                                                           host=request.get_host(),
+                                                           prefix=settings.CELERY_DOWNLOAD_PREFIX,
+                                                           file=str(job_result.get()))
+            return Response({'status': job_result.state, 'result': uri})
+        return Response({'status': job_result.state, 'info': job_result.info})
+
+    def create(self, request, *args, **kwargs):
+        """Create a new CSV export task, using the passed filterkey as a parameter
+
+        filterkey is the same as the "tilekey" that we pass to Windshaft; it must be requested
+        from the Records endpoint using tilekey=true
+        """
+        filter_key = request.data.get('tilekey', None)
+        if not filter_key:
+            return Response({'errors': {'tilekey': 'This parameter is required'}},
+                            status=status.HTTP_400_BAD_REQUEST)
```

Comment:
I'm going to pass on this for now; it makes testing a bit trickier because the ViewSet tries to access Redis, which isn't available on Travis, and I don't have time to put in the necessary mocking to make it work correctly. Hopefully in practice it won't be much of an issue because the polling will pick up the error relatively quickly.